design/0001-automated-releases.md

```diff
@@ -0,0 +1,142 @@
+- Automated Releases
+- @kleimkuhler, @alpeb
+- 2020-02-17 (beginning of sprint)
+- 2020-03-03 (end of sprint)
+- RFC PR: [linkerd/rfc#7](https://github.com/linkerd/rfc/pull/7)
+- Linkerd Issue: [linkerd/linkerd2#3021](https://github.com/linkerd/linkerd2/issues/3021)
+- Reviewers: @grampelberg
+
+# Summary
+
+[summary]: #summary
+
+The goal for this project is to make the Linkerd2 release process as automated
+as possible. The process currently requires that the responsible member take
+multiple manual steps in order for the release artifacts to be properly built,
+tested, and updated for installations.
+
+Instead of addressing these points individually, the release process should be
+re-evaluated as a whole in order to utilize the CI workflow to automate the
+process where possible.
+
+# Problem Statement (Step 1)
+
+[problem-statement]: #problem-statement
+
+The manual steps required in order to perform a release can lead to any number
+of errors by fault of the responsible member. These include errors such as: 
+- forgetting to perform a particular step
+- completing the steps in an incorrect order
+- dealing with flakiness related to the timing of when steps are completed
+
+Additionally, the member must "babysit" the individual steps in order to start
+the next one. This can be time consuming.
+
+Once implemented, the release process will be:
+1. The responsible member opens a pull request that updates `CHANGES.md` with
+   the relevant changes for the upcoming release.
+2. Once step `1` is merged, the member pushes a signed tag for the release.
+3. A workflow is triggered that builds the docker images and tags them with
+   the step `2` release tag. Integration tests are run. If tests pass, a
+   GitHub release and created the the release artifacts are uploaded. The
+   hard-coded Linkerd version in [website](https://github.com/linkerd/website)
+   is automatically updated so that the install script points to the release
+   artifacts.
+4. The member sends out CNCF announcement emails.
+
+# Design proposal (Step 2)
+
+[design-proposal]: #design-proposal
+
+The details of each step are outlined below.
+
+> The responsible member opens a pull request that updates `CHANGES.md` with
+> the relevant changes for the upcoming release.
+
+- The workflows will run unit tests, static checks, and KinD integration tests
+  for this because it continues to trigger pull request conditions
+
+> Step `1` is merged.
+
+- The workflows will be updated to not trigger on `push` events that only edit
+  `.md` files. With this updated condition, the member will not need to wait
+  (or manually cancel) for `master` branch to run through integration tests
```

Comment:
I think we can have both master and branches not run any workflow on solely *.md changes

Reply:
If I understand your comment, "not trigger on `push` events that only edit `.md` files" I am implying that for all branches, not just master.

Reply:
It appeared to me the first part of this piece of text implies the tests will be run on the PR branch:
```
> The responsible member opens a pull request that updates `CHANGES.md` with
> the relevant changes for the upcoming release.
- The workflows will run unit tests, static checks, and KinD integration tests
  for this because it continues to trigger pull request conditions
```

Reply:
Oh yes so the workflows would still be triggered for `pull_request` events. I was thinking that once that PR merges leading to a `push.branches.master` event, the workflows would not run.

Are you suggesting just skip the workflows for both `push` and `pull_request` events that only change `.md` files?

Reply:
Yeah I was suggesting that, but now that you spell it out, I think I take it back :grimacing: 
I remember now we discussed those tests on the PR were the last-moment assurance that let us avoid running them again when the merge happens. And anyways there's not much waiting involved here for the person doing that since usually those tests will finish before the PR gets approved.
So please ignore my comments :slightly_smiling_face:

Reply:
Yep no problem, sounds good!